dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -253,35 +230,123 @@ public TrackedEntity getTrackedEntity(
         trackedEntity.setProgramOwners(filteredProgramOwners);
       }
     } else {
-      // return only tracked entity type attributes
-      TrackedEntityType trackedEntityType = trackedEntity.getTrackedEntityType();
-      if (trackedEntityType != null) {
-        Set<String> tetAttributes =
-            trackedEntityType.getTrackedEntityAttributes().stream()
-                .map(TrackedEntityAttribute::getUid)
-                .collect(Collectors.toSet());
-        Set<TrackedEntityAttributeValue> tetAttributeValues =
-            trackedEntity.getTrackedEntityAttributeValues().stream()
-                .filter(att -> tetAttributes.contains(att.getAttribute().getUid()))
-                .collect(Collectors.toCollection(LinkedHashSet::new));
-        trackedEntity.setTrackedEntityAttributeValues(tetAttributeValues);
+      UserDetails userDetails = CurrentUserUtil.getCurrentUserDetails();
+      trackedEntity =
+          mapTrackedEntity(getTrackedEntityNoDataAcl(uid), params, userDetails, includeDeleted);
+      if (programService.getAllPrograms().stream()
+          .noneMatch(
+              p -> trackerAccessManager.canRead(userDetails, trackedEntity, p, false).isEmpty())) {
+        throw new ForbiddenException(TrackedEntity.class, uid);
```

Comment:
I'd return `ForbiddenException` instead of `NotFoundException` to avoid breaking the existing API.